api_compatibility_policy.md

```diff
@@ -150,10 +152,51 @@ which includes the earliest date each feature will be removed.
-would need to follow the policy on [backwards incompatible changes](#backwards-incompatible-changes),
-but a change that renames the Go struct type for that field is allowed.
+### Covered parts of the codebase
+
+The compatibility policy for the [github.com/tektoncd/pipeline Go package](https://pkg.go.dev/github.com/tektoncd/pipeline) currently only covers
+Go structs representing CRDs in `pkg/apis`, the generated client libraries in `pkg/client`, and the [resolver interface](https://github.com/tektoncd/pipeline/blob/main/pkg/resolution/resolver/framework/interface.go).
+Backwards compatibility for other parts of the codebase is on a best-effort basis.
+
+Pipelines contributors are working on modularizing `pkg/apis`, so that clients only need to import a minimal set of dependencies.
+This work is tracked in [issue #6483](https://github.com/tektoncd/pipeline/issues/6483).
+When it is complete, the compatibility policy will be updated to include all of `pkg/apis`.
+
+Pipelines contributors also plan to move all internal functionality in the entire [github.com/tektoncd/pipeline Go package](https://pkg.go.dev/github.com/tektoncd/pipeline)
+into internal packages. This work is tracked in [issue #5679](https://github.com/tektoncd/pipeline/issues/5679).
+When it is complete, the compatibility policy will be updated to include all of `github.com/tektoncd/pipeline`.
+
+### Alpha CRDs
+
+- Go structs representing alpha CRDs may change in breaking ways at any time. If the change impacts the yaml/JSON API, we will provide one release of warning.
+
+- If support for an alpha CRD is dropped, its client library will not be removed until the last release with support for the CRD has reached its end of life.
+
+- If support for a field is dropped, the field may also be removed from the client libraries; i.e. it is not guaranteed to be tombstoned.
+
+### Beta CRDs
+
+- Go structs representing beta CRDs may change in breaking ways if the yaml/JSON API is unaffected.
+
+- If support for a beta CRD is dropped, its client library will not be removed until the last release with support for the CRD has reached its end of life.
+
+- If support for an alpha field is dropped, the field will be tombstoned in client libraries and will not be removed until the last release with support for the field.
+
+### Stable CRDs
+
+- Go structs representing stable fields may not change in breaking ways without a semver major version bump, even if the yaml/JSON API is unaffected.
+
+- If support for a stable CRD is dropped (via a semver major version bump), the client libraries for this CRD will continue to be supported.
+
+- If support for an alpha field is dropped, the field will be tombstoned in client libraries and will not be removed.
+
+- If backwards incompatible changes are made to an alpha or beta field after the warning period, the same changes will be made to the client libraries at that time. 
```

Comment:
These parts I'm a bit unsure of but it seemed like the best way to achieve consistency with our policies on alpha/beta features-- open to suggestions.